### DISCUSSION TITLE
Printing Compute Graph to Validate Dependencies of Operations

### DISCUSSION THREAD

**Discussion by mjkpolo** [ggml-org/ggml#889] [Q&A] [answered]
I was wondering if there is a way to print the compute graph built by llama.cpp to validate the dependencies of certain operations. Specifically, it's not immediately clear to me when the `ggml_cpy` operations are performed since their result is passed to `ggml_build_forward_expand` and not another `ggml_tensor` so I would like to validate copies into a view of a tensor is done before that tensor is used.

Thanks!!

**Comment by Green-Sky** [comment, marked as answer]
see https://github.com/ggerganov/ggml/issues/882
